### ISSUE TITLE
Add Early stopping

### ISSUE THREAD

**Issue by gcroci2** [DeepRank/deeprank-core#216]
Early stopping is a form of regularization used to avoid overfitting the training dataset. Early stopping keeps track of the validation loss; if the loss stops decreasing for several epochs in a row the training stops. We should have something like [this](https://github.com/Bjarten/early-stopping-pytorch).
